website/docs/sdk-reference/js.md

```diff
@@ -378,89 +468,101 @@ ConfigCat - INFO - Evaluate 'isPOCFeatureEnabled'
+## `getAllValuesAsync()`
 
 Evaluates and returns the values of all feature flags and settings. Passing a [User Object](#user-object) is optional.
 
 ```js
-const configCatClient = configcat.createClient('#YOUR-SDK-KEY#');
+const configCatClient = configcat.getClient('#YOUR-SDK-KEY#');
 
-configCatClient.getAllValues(function (settingValues) {
-  settingValues.forEach((i) =>
-    console.log(i.settingKey + ' -> ' + i.settingValue),
-  );
-});
+let settingValues = await configCatClient.getAllValuesAsync();
+settingValues.forEach((i) =>
+  console.log(i.settingKey + ' -> ' + i.settingValue),
+);
 
 // invoke with user object
-const userObject = { identifier: 'john@example.com' };
-configCatClient.getAllValues(function (settingValues) {
-  settingValues.forEach((i) =>
-    console.log(i.settingKey + ' -> ' + i.settingValue),
-  );
-}, userObject);
+const userObject = new configcat.User('john@example.com');
+
+settingValues = await configCatClient.getAllValuesAsync(userObject);
+settingValues.forEach((i) =>
+  console.log(i.settingKey + ' -> ' + i.settingValue),
+);
 ```
 
+## `getAllValueDetailsAsync()`
+
+Evaluates and returns the values along with evaluation details of all feature flags and settings. Passing a [User Object](#user-object) is optional.
+
 ```js
-const configCatClient = configcat.createClient('#YOUR-SDK-KEY#');
+const configCatClient = configcat.getClient('#YOUR-SDK-KEY#');
 
-const settingValues = await configCatClient.getAllValuesAsync();
-settingValues.forEach((i) =>
-  console.log(i.settingKey + ' -> ' + i.settingValue),
+let settingValues = await configCatClient.getAllValueDetailsAsync();
+settingValues.forEach((details) =>
+  console.log(details),
 );
 
 // invoke with user object
-const userObject = { identifier: 'john@example.com' };
-const settingValuesTargeting = await configCatClient.getAllValuesAsync(
-  userObject,
-);
-settingValuesTargeting.forEach((i) =>
-  console.log(i.settingKey + ' -> ' + i.settingValue),
+const userObject = new configcat.User('john@example.com');
+
+settingValues = await configCatClient.getAllValueDetailsAsync(userObject);
+settingValues.forEach((details) =>
+  console.log(details),
 );
 ```
 
 ## Using custom cache implementation
 
-Config's data stored in a cache, it is efficiency increases in retrieval of data and performance of the client. If you would like to use your cache solution (for example your system uses external or distributed cache) you can implement those function and set to `cache` parameters in the setting.
+Config data is stored in a cache for reducing network traffic and for improving performance of the client. If you would like to use your own cache solution (for example when your system uses external or distributed cache) you can implement the [`ICache` interface](https://github.com/configcat/common-js/blob/master/src/Cache.ts) and set the `cache` parameter in the options.
 
 ```js
-function myCustomCache() {}
+class MyCustomCache {
+  set(key, config) {
```

Comment:
This 'config' might be a chance for a misunderstanding. As I'm not sure if this is some kind of configuration of the cache or is this the config that's being cached. I suggest using config only when trying to reference the config.json.

Reply:
There is a comment a few lines below which describes the meaning of this parameter, so I wouldn't change this unless you insist.